### ISSUE TITLE
Rancher-compose Port Mapping with IP Address Doesn't Work

### ISSUE THREAD

**Issue by sjiveson** [rancher/rancher#6559]
**Rancher Version:**

v1.01 (rancher-compose v0.7.4) and v1.1.0 (rancher-compose v0.8.5)

**Docker Version:**

v1.10.3 and v1.12.1

**OS and where are the hosts located? (cloud, bare metal, etc):**

Rancher OS on bare metal.

v0.4.4 and v0.6.1

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single-node. External DB (same host)

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle

**Steps to Reproduce:**

Specify a host IP address as well as a port in a `port:` mapping in a compose file. Run `rancher-compose` to create and run the service.

Example compose:

```
envoy-fe:
  ports:
    - 10.11.12.116:8080:8080
    - 10.11.12.116:8081:444
  labels:
```

**Results:** 

The port is bound to all host IP addresses, not just the one specified.

The output of `docker ps` confirms this:

`0.0.0.0:8080->8080/tcp, 0.0.0.0:8081->444/tcp`

Ditto the output of `docker inspect` and the `ss` command.

**Expected:**

The port is bound only to the specified address 10.11.12.116 on the host.

This is the behaviour I see when doing a manual `docker run` or using `docker-compose` on my local host.

**Comment by vincent99** [comment]
@sjiveson I think this was fixed somewhere in the 1.2 cycle, but bindings a service to specific IPs is of limited utility since you have to also control the scheduling so that the container goes to exactly the right host that had that IP..

**Comment by sjiveson** [comment]
OK, great, that's good to know. It's essential for hosting multiple TLS protected sites on standard ports where SNI isn't an option.

In our use case, scheduling isn't an issue as we have one container per host, each binding to the same IP:port combo and use keepalived to actually assign the (V)IP to only one host. And move it when necessary etc. etc.

**Comment by vincent99** [comment]
1.4 has support for defining multiple IPs on hosts and scheduling a port to an available IP.

**Comment by sjiveson** [comment]
Thanks @vincent99.

Just for the benefit of others, I've confirmed the issue is resolved in v1.2.
